### ISSUE TITLE
Hardlinked files are not counted properly

### ISSUE THREAD

**Issue by Freaky** [Byron/dua-cli#36]
The same file can appear in multiple places in a directory hierarchy.  This includes things like Rust's own build directories, `/rescue` partitions with tools like busybox, backups using rsync `--link-dest`, and so forth.

Traditional `du`'s deal with this by keeping a map of seen inodes and their link counts, and avoid counting additional copies of the same inode - decrementing the link count until all have been seen and the inode can be forgotten.  This may not be perfect for an interactive tool like this, but it would probably be better than nothing.

**Comment by Byron** [comment]
Thanks a lot for posting your insights on how this is actually implemented in `du`. This might be helpful when trying to resolve this issue.

Performance-wise it might be OK to implement it, even though the memory consumption might go up visibly. However, being correct AND fast (enough) by default is my preference, too, and as long as hardlink tracking can be disabled I would be happy to see this fixed.

**Comment by Freaky** [comment]
```shell
-% target/debug/dua -l /rescue
   1.14 GB /rescue
-% target/debug/dua /rescue
  14.22 MB /rescue
```

:)

**Comment by Byron** [comment]
**cannot** wait :)))!

**Comment by Freaky** [comment]
Hang on while I beat git into submission and repaint part of the bikeshed :)

**Comment by Byron** [comment]
Closed with release v2.3.0 :)!
